include/grpcpp/ext/server_metric_recorder.h

```diff
@@ -27,11 +27,9 @@
 #include "absl/strings/string_view.h"
 #include "absl/types/optional.h"
 
-#include <grpcpp/impl/sync.h>
+#include "src/core/ext/filters/client_channel/lb_policy/backend_metric_data.h"
```

Comment:
Actually I just added this include without knowing the above, fixed.

But then now we have a struct of { BackendMetricData, uint64_t } and I could not compile. Instead I moved this next to BackendMetricData. Let me know if this looks OK.

Reply:
Updated, I *think* I went with the option 3 you suggested.

But now all tests are stuck and I still could not find why. It's stuck in grpc_init() likely due to a deadlock holding the global lock, but I have no idea why this happened. Thoughts?

Reply:
I moved BackendMetricDataState into backend_metric_recorder.h instead, I expect this is fine as it's for the server side only.  This way we don't need to refactor only to bring it in the definition of ServerMetricRecorder.

Reply:
I think the way you had this looked basically right.  You might need to have `ServerMetricRecorderImpl` also inherit from [`internal::GrpcLibrary`](https://github.com/grpc/grpc/blob/54ad01cf01696a614df8b2256a134aa561ed7315/include/grpcpp/impl/grpc_library.h#L32) to make sure `grpc_init()` gets handled right.

I'd prefer that approach if possible, because it avoids exposing the internal implementation details in the header file at all, even as private.  But if you can't make that work for some reason, I can live with the way you have this now, as long as we preserve the option to switch to the other approach later.  Specifically, please make the ctor private and add a factory method, so that we can later change this to be an abstract interface if we want to:

```
class ServerMetricRecorder {
 public:
  static std::unique_ptr<ServerMetricRecorder> Create();

  // ...

 private:
  ServerMetricRecorder();

  // ...
};
```

Reply:
I went back and `internal::GrpcLibrary` but it still timed out, unfortunately. Given the timeline I think we can go with the current option. I made the ctor private.